src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -52,10 +52,10 @@ public sealed class RSessionProvider : IRSessionProvider {
         public event EventHandler<BrokerStateChangedEventArgs> BrokerStateChanged;
         public event EventHandler<HostLoadChangedEventArgs> HostLoadChanged;
 
-        public RSessionProvider(ICoreServices services, IConsole callback = null) {
+        public RSessionProvider(ICoreShell coreShell, IConsole callback = null) {
```

Comment:
One of the reasons why I wanted to use `microsoft.extensions.dependencyinjection` is to be able to supply services directly into constructor and avoid managing dependencies directly.